_posts/2019-02-09-snowplow-scala-analytics-sdk-0.4.0-released/index.md

```diff
@@ -0,0 +1,105 @@
+---
+layout: post
+title: Snowplow Scala Analytics SDK 0.4.0 released
+title-short: Snowplow Scala Analytics SDK 0.4.0
+tags: [scala, snowplow, enriched events, spark, dynamodb]
+author: Rostyslav
+category: Releases
+permalink: /blog/2019/02/09/snowplow-scala-analytics-sdk-0.4.0-released/
+---
+
+We are excited to announce the 0.4.0 release of the [Snowplow Scala Analytics SDK][sdk-repo], a library that provides tools to process and analyze Snowplow enriched events in [Apache Spark][spark], [AWS Lambda][lambda], [Apache Flink][flink], [Scalding][scalding] and other JVM-compatible data processing frameworks. This release reworks the JSON Event Transformer to use a new type-safe API, and introduces several other internal changes.
+
+Read on below the fold for:
+
+1. [Event API](#event-api)
+2. [Using the typesafe API](#typesafe-api)
+3. [Additional changes](#additional-changes)
+4. [Upgrading](#upgrading)
+5. [Getting help](#help)
+
+<!--more-->
+
+<h2 id="event-api">1. Event API</h2>
+
+Previously, the JSON Event Transformer - an event transformer that takes a Snowplow enriched event and converts it into a JSON ready for further processing - used to return Strings, which represented enriched events turned into JSON objects. While this was a very unopinionated and minimalistic approach, it involved a lot of extra post-processing, namely:
+
+- Accessing individual JSON fields required casting the result to an instance of the json4s AST class via unsafe operations, `parse(result)` or even `parse(result).asInstanceOf[JObject]`.
+- Accessing always existing fields sill required redundant error processing logic, e.g. `parsedJson.map("event_id").getOrElse(throw new RuntimeException("event_id is not present in the enriched event")`
+- Getting a list of shredded types required having to use a separate function, `jsonifyWithInventory`.
```

Comment:
This sentence doesn't make sense to me. Is this still correct?
> _To get_ a list of _the_ shredded types required _you need_ to use a separate function, `jsonifyWithInventory`.

Reply:
It's no longer correct. I've reworded it as

>To get a list of shredded types a separate function, `jsonifyWithInventory`, had to be used.